bundles/org.openhab.binding.dwdpollenflug/src/main/java/org/openhab/binding/dwdpollenflug/internal/handler/DWDPollenflugRegionHandler.java

```diff
@@ -0,0 +1,162 @@
+            } else {
+                updateStatus(ThingStatus.ONLINE);
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+        }
+    }
+
+    private synchronized @Nullable DWDPollenflugBridgeHandler syncToBridge() {
+        if (bridgeHandler == null) {
+            Bridge bridge = getBridge();
+            if (bridge == null) {
+                return null;
+            }
+            ThingHandler handler = bridge.getHandler();
+            if (handler instanceof DWDPollenflugBridgeHandler) {
+                bridgeHandler = (DWDPollenflugBridgeHandler) handler;
+                bridgeHandler.registerRegionListener(this);
+            } else {
+                return null;
+            }
+        }
+
+        return bridgeHandler;
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("DWDPollenflug region handler disposes. Unregistering listener.");
+        DWDPollenflugBridgeHandler handler = syncToBridge();
+        if (handler != null) {
+            handler.unregisterRegionListener(this);
+            bridgeHandler = null;
+        }
+    }
+
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
+        if (!ignoreConfigurationUpdate) {
+            super.handleConfigurationUpdate(configurationParameters);
+        }
+    }
+
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
+        DWDPollenflugBridgeHandler handler = syncToBridge();
+        if (handler != null) {
+            DWDPollenflug pollenflug = handler.getPollenflug();
+            if (pollenflug != null) {
+                notifyOnUpdate(pollenflug);
+            }
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void notifyOnUpdate(DWDPollenflug pollenflug) {
+        DWDRegion region = pollenflug.getRegion(thingConfig.getRegionID());
+        if (region == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Region not found");
+            return;
+        }
+
+        updateStatus(ThingStatus.ONLINE);
+
+        ignoreConfigurationUpdate = true;
+        updateProperties(region.getProperties());
+        ignoreConfigurationUpdate = false;
+
+        updateChannels(region.getChannels());
+
+        updateChannels(pollenflug.getChannels());
+
+        if (lastUpdate == null || !lastUpdate.equals(pollenflug.getLastUpdate())) {
+            triggerChannel(CHANNEL_UPDATES + "#" + CHANNEL_UPDATED, TRIGGER_REFRESHED);
+            lastUpdate = pollenflug.getLastUpdate();
+        }
+    }
+
+    private void updateChannels(Map<String, State> channels) {
+        for (Entry<String, State> entry : channels.entrySet()) {
+            String channelID = entry.getKey();
+            if (isLinked(channelID)) {

```

Comment:
no need to check `isLinked`, it's a very expensive call. Let the framework take care of that.

Reply:
Please check my fix!